### PR TITLE
{tools}[foss-2015a] Subversion 1.8.14 (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**
Needs ~~https://github.com/hpcugent/easybuild-easyblocks/pull/689~~ (Serf EasyBlock)
Inspired by PR #758 

edit: now requires https://github.com/hpcugent/easybuild-easyblocks/pull/700

### CODE REVIEW

File: easybuild/easyconfigs/s/Serf/Serf-1.3.8-foss-2015a.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+]
+
+builddependencies = [('SCons', '2.3.6', '-Python-2.7.9')]
+
+dependencies = [
+    ('APR', '1.5.2'),
+    ('APR-util', '1.5.4'),
+]
+
+osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel')]
```
Comment (wpoely86) [inline easybuild/easyconfigs/s/Serf/Serf-1.3.8-foss-2015a.eb]: also add a commented out openssl to dependencies, like in the Python easyconfig

**Comment by wpoely86** [comment]
lgtm

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 4 out of 6 (5 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/bb15d2b3ce2b799d5141 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 4 out of 6 (5 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/9d818ea7b8386c76f3b7 for a full test report.

**Comment by boegel** [comment]
Hmm, failing unit tests in Serf?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 3 out of 4 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/91a4b9d1a9d7f86f8e16 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 3 out of 4 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/a33cad646ee61197d91b for a full test report.

**Comment by wpoely86** [comment]
if I run them manually, it works...

**Comment by boegel** [comment]
@wpoely86: it's unclear to me what's failing exactly...

**Comment by wpoely86** [comment]
@boegel yes, they don't bother showing the output of tests

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/643af7bc5181bbd4a7b9 for a full test report.

**Comment by boegel** [comment]
@wpoely86: what changed?

**Comment by wpoely86** [comment]
@boegel nothing. I didn't run it inside a jobscript. That's the only difference...

**Comment by boegel** [comment]
ulimit? cpusets?

**Comment by wpoely86** [comment]
Possible, it does tcp/ip stuff. @hajgato can you also submit a test report? So we at least know if it is reproducible.

File: easybuild/easyconfigs/s/Subversion/Subversion-1.8.14-foss-2015a.eb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    'http://www.us.apache.org/dist/%(namelower)s',
+]
+sources = [SOURCELOWER_TAR_BZ2]
+
+dependencies = [
+    ('APR', '1.5.2'),
+    ('APR-util', '1.5.4'),
+    ('SQLite', '3.8.8.1'),
+    ('zlib', '1.2.8'),
+    ('Serf', '1.3.8'),
```
Comment (boegel) [inline easybuild/easyconfigs/s/Subversion/Subversion-1.8.14-foss-2015a.eb]: @hajgato: I also had `neon` as a dep in #758, any reason you dropped it?

File: easybuild/easyconfigs/s/SCons/SCons-2.3.6-foss-2015a-Python-2.7.9.eb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+homepage = 'http://www.scons.org/'
+description = "SCons is a software construction tool."
+
+toolchain = {'name': 'foss', 'version': '2015a'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://prdownloads.sourceforge.net/%(namelower)s']
+
+python = 'Python'
+pyver = '2.7.9'
```
Comment (boegel) [inline easybuild/easyconfigs/s/SCons/SCons-2.3.6-foss-2015a-Python-2.7.9.eb]: 2.7.10?

File: easybuild/easyconfigs/s/Serf/Serf-1.3.8-foss-2015a.eb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+source_urls = ['https://archive.apache.org/dist/%(namelower)s']
+sources = [SOURCELOWER_TAR_BZ2]
+
+patches = [
+    ('%(name)s-%(version)s_remve_c99_comment.patch'),
+    ('%(name)s-%(version)s_SCons_ld_lib.patch'),
+]
+
+builddependencies = [('SCons', '2.3.6', '-Python-2.7.9')]
```
Comment (boegel) [inline easybuild/easyconfigs/s/Serf/Serf-1.3.8-foss-2015a.eb]: Python-2.7.10?

File: easybuild/easyconfigs/s/Subversion/Subversion-1.8.14-foss-2015a.eb (hunk 2 above)
Comment (hajgato) [inline easybuild/easyconfigs/s/Subversion/Subversion-1.8.14-foss-2015a.eb]: neon is replaced by Serf in svn 1.8 (neon is supported up to 1.6, Serf from 1.8) svn 1.7 is nuked.

Comment (boegel) [inline easybuild/easyconfigs/s/Subversion/Subversion-1.8.14-foss-2015a.eb]: ok, thanks for clarifying that

**Comment by hajgato** [comment]
Test report by @hajgato
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.6.6
See https://gist.github.com/22aefff6f7c47100f248 for a full test report.

**Comment by hajgato** [comment]
what about internet access? It uses some special ports, like 8080 and others. (do not know exactly)

**Comment by hajgato** [comment]
Test report by @hajgato
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux RHEL 6.3, AMD Opteron(TM) Processor 6272, Python 2.6.6
See https://gist.github.com/dd1bec64935537acbd60 for a full test report.

**Comment by boegel** [comment]
@hajgato: not sure what you mean with your remark on pot 8080?

**Comment by hajgato** [comment]
@boegel: the reason why Serf test could fail. It does some internet connection tests.

**Comment by boegel** [comment]
@hajgato: I don't think that was the issue, our workernodes have internet access

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 12 out of 13 (7 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/58c0358d28469428e414 for a full test report.

**Comment by boegel** [comment]
@wpoely86: failing tests again...

**Comment by wpoely86** [comment]
@boegel okay, it's something on our clusters in the build env. No idea what.

**Comment by boegel** [comment]
@wpoely86: testing it myself now, in a job

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 20 out of 20 (7 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/6c06411b7ffba2b5eaa3 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 20 out of 20 (7 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/6e1f6f2607f348ef26f2 for a full test report.

**Comment by boegel** [comment]
@wpoely86: works fine for me, even when the test is done from a job... must be something specific in your account? would be interesting to figure out what's causing the problem...

**Comment by wpoely86** [comment]
@boegel give me a dump of your environment in the jobscript?

**Comment by wpoely86** [comment]
Might be cgroups. I run it on 4 cores and yours is on a full node.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 6 out of 7 (7 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/c91aee0f6ff125e3a976 for a full test report.

**Comment by boegel** [comment]
@wpoely86: ok, spot on it seems, issue reproduced with 4-core job... now what? :)

**Comment by wpoely86** [comment]
If you really want to know, you have to isolate the tests that fail and look what they do. Lots of digging.

**Comment by boegel** [comment]
@wpoely86: not worth the trouble imho, we should just disable to tests (see also https://github.com/hpcugent/easybuild-easyblocks/pull/689/files#r40216888)

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 13 out of 13 (7 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/f74b3a6b2d19dd551854 for a full test report.

**Comment by boegel** [comment]
https://github.com/hpcugent/easybuild-easyblocks/pull/700 should get merged first, so we can disable the broken test bit

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/10088c3208c770f44b90 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2c2ffe0597a9d9d91485 for a full test report.

**Comment by boegel** [comment]
Good to go, thanks @hajgato!

**Pull request merged**